### ISSUE TITLE
Changing selected file in file browser is slow

### ISSUE THREAD

**Issue by krassowski** [jupyterlab/jupyterlab#16767]
## Description

It can take a couple of seconds to select a file down there without mouse by using <kbd>Down</kbd> in folders with hunderds of files

## Reproduce

1. Open JupyterLab node_modules director
2. Select a folder in the the filebrowser
3. Press arrow down for 3 seconds
4. See the delay in the render state

![image](https://github.com/user-attachments/assets/31c6a0fb-227f-4d99-861d-3a2cf39394fc)

Zoomed in:

![image](https://github.com/user-attachments/assets/bf5c5c8a-2ffe-4d52-8b5e-84b0fd81c220)

## Expected behavior

No delay.

Basically we should only refresh two items (the one unselected + the one selected), not the entire listing.

In case of multi-selection we should re-render the changed items only as well.

## Context

- Browser and version: Chrome and Firefox
- JupyterLab version: 4.3.0b1

**Comment by krassowski** [comment]
An alternative solution would be last used cache on renderer (as a function of the item model state)

**Comment by Rishab87** [comment]
Hi @krassowski , can I try solving this, btw I tried to use last used cache on renderer something like this:
```typescript
const cached = this._cache.get(model);
  const currentState = {
    selected: selected ?? false,
    name: model.name,
    fileType,
    size: model.size
};

if (cached && this._isStateEqual(cached.lastState, currentState)) {
  // If the state hasn't changed, we don't need to update anything
  return;
} 
```
inside updateItemNode function in listings.ts in packages/filebrowser but this approach doesn't seem to work as I'm still experiencing delays in filebrowser, am I referring to the correct part of the code?
